src/main/java/org/opensrp/service/PhysicalLocationService.java

```diff
@@ -71,7 +72,33 @@ public void update(PhysicalLocation physicalLocation) {
 		if (StringUtils.isBlank(physicalLocation.getId()))
 			throw new IllegalArgumentException("id not specified");
 		physicalLocation.setServerVersion(null);
-		locationRepository.update(physicalLocation);
+		PhysicalLocation existingEntity = locationRepository.findLocationByIdentifierAndVersion(physicalLocation.getId(), physicalLocation.getProperties().getVersion());
+		boolean locationHasNoUpdates = locationRepository.isGeometryCoordsEqual(physicalLocation, existingEntity);
+		if (locationHasNoUpdates || !physicalLocation.isJurisdiction()){
+			locationRepository.update(physicalLocation);
+		} else {
+			//make existing location inactive
+			physicalLocation.getProperties().setStatus(LocationProperty.PropertyStatus.INACTIVE);
+			locationRepository.update(physicalLocation);
+
+			// create new location
+			PhysicalLocation newPhysicalLocation = new PhysicalLocation();
+			newPhysicalLocation.setGeometry(physicalLocation.getGeometry());
+			newPhysicalLocation.setId(physicalLocation.getId());
+			newPhysicalLocation.setJurisdiction(physicalLocation.isJurisdiction());
+			newPhysicalLocation.setLocationTags(physicalLocation.getLocationTags());
+			newPhysicalLocation.setType(physicalLocation.getType());
+			newPhysicalLocation.setServerVersion(null);
+			LocationProperty newLocationProperty = physicalLocation.getProperties();
+			//increment location version
+			int newVersion = physicalLocation.getProperties().getVersion() + 1;
+			newLocationProperty.setVersion(newVersion);
+			newLocationProperty.setStatus(LocationProperty.PropertyStatus.ACTIVE);
+			newPhysicalLocation.setProperties(newLocationProperty);
```

Comment:
This is not necessary since `physicalLocation` has the location passed should be saved/updated You only need to increment the version, which should be computed from the previous version +1